src/IssueViz.UnitTests/Editor/LocationTagging/LocationTaggerTests.cs

```diff
@@ -0,0 +1,431 @@
+
+            testSubject.TagSpans.Should().NotBeSameAs(initialTags);
+        }
+
+        [TestMethod]
+        public void GetTags_NoTags_ReturnsEmpty()
+        {
+            var testSubject = new LocationTagger(ValidBuffer, ValidStore, ValidSpanCalculator, ValidLogger);
+
+            var validSpan = new Span(0, ValidBuffer.CurrentSnapshot.Length);
+            var inputSpans = new NormalizedSnapshotSpanCollection(ValidBuffer.CurrentSnapshot, validSpan);
+
+            testSubject.GetTags(inputSpans)
+                .Should().BeEmpty();
+        }
+
+        [TestMethod]
+        public void GetTags_EmptyInputSpan_ReturnsEmpty()
+        {
+            var testSubject = new LocationTagger(ValidBuffer, ValidStore, ValidSpanCalculator, ValidLogger);
+
+            var inputSpans = new NormalizedSnapshotSpanCollection();
+
+            testSubject.GetTags(inputSpans)
+                .Should().BeEmpty();
+        }
+
+        [TestMethod]
+        public void GetTags_NoOverlappingSpans_ReturnsEmpty()
+        {
+            // Make sure the tagger and NormalizedSpanCollection use the same snapshot
+            // so we don't attempt to translate the spans
+            var snapshot = CreateSnapshot(length: 40);
+            var locSpan1 = new Span(1, 10);
+            var locSpan2 = new Span(11, 10);
+            var storeMock = CreateStoreWithLocationsWithSpans(snapshot, locSpan1, locSpan2);
+
+            var inputSpans = new NormalizedSnapshotSpanCollection(snapshot, new Span(32, 5));
+
+            var testSubject = new LocationTagger(ValidBuffer, storeMock, ValidSpanCalculator, ValidLogger);
+            testSubject.TagSpans.Count().Should().Be(2); // sanity check
+
+            var matchingTags = testSubject.GetTags(inputSpans);
+            matchingTags.Should().BeEmpty();
+        }
+
+        [TestMethod]
+        public void GetTags_HasOverlappingSpans_ReturnsExpectedTags()
+        {
+            // Make sure the tagger and NormalizedSpanCollection use the same snapshot
+            // so we don't attempt to translate the spans
+            var snapshot = CreateSnapshot(length: 40);
+            var locSpan1 = new Span(1, 10);
+            var locSpan2 = new Span(11, 10);
+            var locSpan3 = new Span(21, 10);
+            var storeMock = CreateStoreWithLocationsWithSpans(snapshot, locSpan1, locSpan2, locSpan3);
+
+            var inputSpans = new NormalizedSnapshotSpanCollection(snapshot, new Span(18, 22));
+
+            var testSubject = new LocationTagger(ValidBuffer, storeMock, ValidSpanCalculator, ValidLogger);
+            testSubject.TagSpans.Count().Should().Be(3); // sanity check
+
+            var matchingTags = testSubject.GetTags(inputSpans);
+            matchingTags.Select(x => x.Span.Span).Should().BeEquivalentTo(locSpan2, locSpan3);
+        }
+
+        [TestMethod]
+        public void GetTags_DifferentSnapshots_SnapshotsAreTranslated()
```

Comment:
I worked out how to mock enough of the snapshot infrastructure to be able to write tests for "TranslateTo".
Essentially, each snapshot has an `ITextVersion` object which points to the `Next` `version object, and contains a list of changes between the version. I created versions with an empty changes collection which seemed to do the trick.